internal/encryption/pkcs11.go

```diff
@@ -0,0 +1,194 @@
+package encryption
+
+import (
+	"crypto/rand"
+	"fmt"
+
+	"github.com/miekg/pkcs11"
+)
+
+// PKCS11Backend implements EncryptionBackend using PKCS11 protocol for HSMs.
+type PKCS11Backend struct {
+	ctx   *pkcs11.Ctx
+	pin   string
+	keyID uint16
+}
+
+const ivSize = 16
+
+// NewPKCS11Backend creates a new PKCS11Backend.
+func NewPKCS11Backend(libPath string, pin string, keyID uint16) (*PKCS11Backend, error) {
+	ctx := pkcs11.New(libPath)
+	if ctx == nil {
+		return nil, fmt.Errorf("failed to load PKCS#11 library at %s", libPath)
+	}
+
+	return &PKCS11Backend{
+		ctx:   ctx,
+		pin:   pin,
+		keyID: keyID,
+	}, nil
+}
+
+// Encrypt encrypts the plaintext using AES-CBC algorithm.
+func (h *PKCS11Backend) Encrypt(data []byte) ([]byte, error) {
+	session, key, err := h.connectToBackend()
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to backend: %w", err)
+	}
+	defer func() {
+		if cleanupErr := h.cleanupSession(session); cleanupErr != nil {
```

Comment:
Can't we use the logger instead of `fmt.Printf`? The same comment may apply elsewhere

Reply:
Replaced with logger.